web/chat/chat-tabs.react.js

```diff
@@ -0,0 +1,56 @@
+// @flow
+
+import * as React from 'react';
+
+import ChatThreadTab from './chat-thread-tab.react';
+import css from './chat-tabs.css';
+import ChatThreadHome from './chat-thread-home.react';
+import ChatThreadBackground from './chat-thread-background.react';
+
+type Props = {||};
+type State = {|
+  activeTab: string,
+|};
+
+class ChatTabs extends React.PureComponent<Props, State> {
+  state = {
+    activeTab: 'HOME',
+  };
+
+  onClickTabItem = (tab: string) => {
+    this.setState({ activeTab: tab });
+  };
+  homeTab = (
```

Comment:
It seems that `homeTab` and `backgroundTab` are initialized when the class is constructed, and don't get redefined if `this.state.activeTab` changes. Does the underline actually move when you press a button?

I don't think they should be class fields, we can probably just define them in `render`. That also would allow us to skip the `key` prop.

```js
    return (
      <div className={css.container}>
        <ol>
          <ChatThreadTab
            activeTab={this.state.activeTab}
            title="HOME"
            onClick={this.onClickTabItem}
          />
          <ChatThreadTab
            activeTab={this.state.activeTab}
            title="BACKGROUND"
            onClick={this.onClickTabItem}
          />
        </ol>
        <div>{threadList}</div>
      </div>
    );
```

Reply:
yeah, you're totally right, underline is not moving. At first I defined them in `render`, then changed do class fields, but somehow missed that pressing the button is not changing underline. Sorry about that!